payment-in-stablecoins/README.md

```diff
@@ -0,0 +1,252 @@
+- `maxAdminBotDiscountBPs` The maximum value for `botDiscountBPs` that `admin` is allowed to set
+
+#### View functions
+
+`function ethNeeded(uint256 additionalTokens, uint256 bufferBPs) public view returns (uint256)`
+
+- Get how much ETH this contract needs in order to fund its current obligations plus `additionalTokens`, with a safety buffer `bufferBPs` basis points
+
+`function tokenAmountNeeded() public view returns (uint256)`
+
+- Returns the amount of tokens this contract is willing to exchange of ETH
+  - zero if it has enough tokens
+  - otherwise `baselinePaymentTokenAmount + payer.totalDebt() - paymentToken.balanceOf(address(payer))`
+
+`function price() public view returns (uint256)`
+
+- Returns the ETH/`paymentToken` price this contract is willing to exchange ETH at, including the discount
+
+`function ethAmountPerTokenAmount(uint256 tokenAmount) public view returns (uint256)`
+
+- Returns the amount of ETH this contract will send in exchange for `tokenAmount` tokens
+
+`function tokenAmountNeededAndETHPayout() public view returns (uint256, uint256)`
+
+- Returns the amount of tokens the contract can buy and the amount of ETH it will pay for it
+
+`function tokenAmountPerEthAmount(uint256 ethAmount) public view returns (uint256)`
+
+- Returns the amount of tokens the contract expects in return for eth
+
+#### `buyETH` transaction
+
+##### Description
+
+- Buy ETH from this contract in exchange for `paymentToken` tokens
+- The price is determined using `priceFeed` plus `botDiscountBPs`
+- Immediately invokes `payer` to pay back outstanding debt
+
+##### Rules
+
+`function buyETH(uint256 tokenAmount)`
+
+- `msg.sender` must approve this contract to spend `amount` of stablecoin, which it will transfer to `Payer`
+- should revert if contract has insufficient ETH balance
+
+`function buyETH(uint256 tokenAmount, address to, bytes calldata data)`
+
+- `to` must implement the interface `IBuyETHCallback.buyETHCallback`
+- as part of `buyETHCallback`, sender must transfer `amount` stablecoins to `Payer`
+- should revert if contract has insufficient ETH balance
+
+#### Admin / Owner transactions
+
+##### General Rules
+
+- `msg.sender` must be `admin` or `owner`
+
+##### Specific Rules
+
+`function setBotDiscountBPs(uint16 newBotDiscountBPs)`
+
+- if `msg.sender == admin`, the new value must be within the min/max bounds set by `owner`
+
+`function setBaselinePaymentTokenAmount(uint256 newBaselinePaymentTokenAmount)`
+
+- if `msg.sender == admin`, the new value must be within the min/max bounds set by `owner`
+
+`function withdrawETH()`
+
+- the contract's ETH balance must be sent to `owner`
+- should revert if the transfer fails
+
+`function pause()`
+`function unpause()`
+`function setAdmin(address newAdmin)`
+
+- no additional rules
+
+#### Owner-only transactions
+
+##### Rules
+
+- `msg.sender` must be `owner`
+
+`function setMinAdminBotDiscountBPs(uint16 newMinAdminBotDiscountBPs)`
```

Comment:
in markdown it's all one line
maybe make a code block

Reply:
thank you, fixed with newlines in between